src/pages/GuestInfo.tsx

```diff
@@ -0,0 +1,159 @@
+      setLoading(true);
+      if (
+        value.firstname === '' ||
+        value.lastname === '' ||
+        value.email === '' ||
+        value.phone === ''
+      ) {
+        throw Error('Empty fields');
+      }
+      if (checkout === undefined) {
+        throw Error('Something went wrong');
+      }
+
+      await axios.post(
+        backend.url + '/offers/' + checkout.pricedOffer.offerId + '/pii',
+        value
+      );
+      dispatch({
+        type: 'SET_CHECKOUT',
+        payload: {
+          ...checkout,
+          personalInfo: value
+        }
+      });
+
+      logger.info('Guest info sent successfully');
+      navigate('/checkout/' + checkout.pricedOffer.offerId);
+    } catch (error) {
+      const message = (error as Error).message || 'Unknown useAuthRequest error';
+      setLoading(false);
+      setError(message);
+    }
+  }, [value]);
+
+  return (
+    <PageWrapper
+      breadcrumbs={[
+        {
+          label: 'Search',
+          path: '/'
+        },
+        {
+          label: 'Facility',
+          path: '/facilities/' + checkout?.facilityId
+        }
+      ]}
+    >
+      <Box align="center" overflow="hidden">
+        <Box width="large" margin="small">
+          <MessageBox loading type="info" show={loading}>
+            Processsing...
+          </MessageBox>
+          <MessageBox type="error" show={!!error}>
+            {error}
+          </MessageBox>
+          <Form
+            onSubmit={() => handleSubmit()}
+            onChange={(newValue) => setValue({ ...value, ...newValue })}
+          >
+            <FormField
+              name="firstname"
+              htmlFor="firstname"
+              label="First Name"
+              validate={{
+                regexp: /^[a-zA-Z]+(?:\s+[a-zA-Z]+)*$/g,
+                message: 'Only a-z, A-Z chars',
+                status: 'error'
+              }}
+            >
+              <TextInput id="firstname" name="firstname" />
+            </FormField>
+            <FormField
+              name="lastname"
+              htmlFor="lastname"
+              label="Last Name"
+              validate={{
+                regexp: /^[a-zA-Z]+(?:\s+[a-zA-Z]+)*$/g,
+                message: 'Only a-z, A-Z chars',
+                status: 'error'
+              }}
+            >
+              <TextInput id="lastname" name="lastname" />
+            </FormField>
+            <FormField name="birthday" htmlFor="birthday" label="Date of birth">
+              <DateInput
+                calendarProps={{
+                  bounds: ['01/01/1900', DateTime.now().toFormat('dd/mm/yyyy')]
+                }}
+                format="dd/mm/yyyy"
+                name="birthday"
+                id="birthday"
+              />
+            </FormField>
+            <FormField
+              name="email"
+              htmlFor="email"
+              label="Email"
+              validate={{
+                regexp: /^[\w-]+@([\w-]+\.)+[\w-]{2,4}$/g,
```

Comment:
are you sure that your regexps are correct? I recommend using regexps from our `@windindingtree/orgid-sdk` because these regesps are tested.

Reply:
✔️

Reply:
import { regexp } from '@windingtree/org.id-utils';

email, phone, birthdate now validated with  `@windingtree/org.id-utils`